File: resc/WebApplication/src/modules/application/models/actual-education.dto.ts
import { NomenclatureDto } from "src/modules/nomenclature/common/models/nomenclature-dto";
import { EducationType } from "../enums/education-type.enum";
import { StudentStatusType } from "../enums/student-status-type.enum";

export class ActualEducationDto {
  educationalQualification: NomenclatureDto;
  status: StudentStatusType | null;
  courseYear: string;
  graduationDate: Date | null;
  educationType: EducationType | null;

  constructor() {
    this.status = StudentStatusType.current;
    this.courseYear = '1';
  }
}

File: resc/Resc.Application/Applications/Dtos/Create/ActualEducationDto.cs
using Resc.Application.Nomenclatures.Dtos;
using Resc.Data.Applications;
using Resc.Data.Applications.Enums;
using Resc.Data.Nomenclatures;
using System;

namespace Resc.Application.Applications.Dtos.Create
{
	public class ActualEducationDto
	{
		public NomenclatureDto<EducationalQualification> EducationalQualification { get; set; }
		public StudentStatusType? Status { get; set; }
		public string CourseYear { get; set; }
		public DateTime? GraduationDate { get; set; }
		public EducationType? EducationType { get; set; }

		public ActualEducation ToModel()
			=> new ActualEducation(this.EducationalQualification.Id, this.Status, this.CourseYear, this.GraduationDate, this.EducationType);
	}
}
